docs/spec/spec.md

```diff
@@ -0,0 +1,202 @@
+| from\*             | ObjectRef      | The originating Channel for the link.                                        | Must be a Channel. |
+| call<sup>1</sup>   | EndpointSpec   | Optional processing on the event. The result of call will be sent to result. |                    |
+| result<sup>1</sup> | ResultStrategy | The continuation for the link.                                               |                    |
+
+\*: Required
+
+1: At Least One(call, result)
+
+#### Metadata
+
+##### Owner References
+
+- If a resource controller created this Subscription: Owned by the originating
+  resource.
+
+##### Conditions
+
+- **Ready.**
+- **FromReady.**
+- **CallActive.** True if the call is sinking events without error.
+- **Resolved.**
+
+#### Events
+
+- PublisherAcknowledged
+- ActionFailed
+
+### Life Cycle
+
+| Action | Reactions                                                        | Limitations |
+| ------ | ---------------------------------------------------------------- | ----------- |
+| Create | The publisher referenced needs to be watching for Subscriptions. |             |
+| Update |                                                                  |             |
+| Delete |                                                                  |             |
+
+---
+
+## kind: ClusterProvisioner
+
+### group: eventing.knative.dev/v1alpha1
+
+_Describes an abstract configuration of a Source system which produces events
+or a Channel system that receives and delivers events._
+
+### Object Schema
+
+#### Spec
+
+| Field      | Type                                                                            | Description                                                                 | Limitations      |
+| ---------- | ------------------------------------------------------------------------------- | --------------------------------------------------------------------------- | ---------------- |
+| type\*     | [GroupKind](https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#GroupKind) | The type of the resource to be provisioned.                                 | Must be Channel. |
+| parameters | runtime.RawExtension (JSON object)                                              | Description of the arguments able to be passed by the provisioned resource. | JSON Schema      |
+
+\*: Required
+
+#### Status
+
+| Field      | Type       | Description                   | Limitations |
+| ---------- | ---------- | ----------------------------- | ----------- |
+| conditions | Conditions | ClusterProvisioner conditions |             |
+
+##### Conditions
+
+- **Ready.**
+
+#### Events
+
+- Resource Created.
+- Resource Removed.
+
+---
+
+## Shared Object Schema
+
+### ProvisionerReference
+
+| Field | Type            | Description | Limitations |
+| ----- | --------------- | ----------- | ----------- |
+| ref\* | ObjectReference |             |             |
```

Comment:
Does this need to be an ObjectReference, or would a `clusterName` be sufficient? I don't think we'd expect to use other objects as Provisioners.

Reply:
that is the object the provisioner made. so it would be ref to a channel. I think ref still helps find it better because tooling is real good around this task.

Reply:
I am sorry, I read "Provisioned" yes you are correct,  BUT, we get into the same jam as bus vs clusterbus if we change this from a ref.